packages/commonwealth/client/scripts/state/api/threads/createThreadToken.ts

```diff
@@ -0,0 +1,81 @@
+import { commonProtocol } from '@hicommonwealth/evm-protocols';
+import { useMutation } from '@tanstack/react-query';
+import TokenLaunchpad from 'helpers/ContractHelpers/tokenLaunchpad';
+
+export interface CreateThreadTokenProps {
+  name: string;
+  symbol: string;
+  threadId: number;
+  initPurchaseAmount: number;
+  chainId: number;
+  walletAddress: string;
+  authorAddress: string;
+  communityTreasuryAddress: string;
+  chainRpc: string;
+  paymentTokenAddress: string;
+  ethChainId: number;
+}
+
+export const createThreadToken = async ({
+  name,
+  symbol,
+  threadId,
+  initPurchaseAmount,
+  chainId,
+  walletAddress,
+  authorAddress,
+  communityTreasuryAddress,
+  paymentTokenAddress,
+  ethChainId,
+  chainRpc,
+}: CreateThreadTokenProps) => {
+  if (
+    !commonProtocol.factoryContracts ||
+    !commonProtocol.factoryContracts[ethChainId] ||
+    !commonProtocol.factoryContracts[ethChainId].factory
```

Comment:
Not all contracts that have a factory will have the `postTokenLaunchpad` or `postTokenBondingCurve`. You can use `mustBeProtocolChainId` function from `chainConfig.ts` to first narrow the type of `chainId` and then verify whether the correct addresses are set.